source/httpfs/partial_driver.go

```diff
@@ -108,9 +108,14 @@ func (p *PartialDriver) Next(version uint) (nextVersion uint, err error) {
 // ReadUp is part of source.Driver interface implementation.
 func (p *PartialDriver) ReadUp(version uint) (r io.ReadCloser, identifier string, err error) {
 	if m, ok := p.migrations.Up(version); ok {
-		body, err := p.fs.Open(path.Join(p.path, m.Raw))
+		upPath := path.Join(p.path, m.Raw)
+		body, err := p.fs.Open(upPath)
 		if err != nil {
-			return nil, "", err
+			return nil, "", &os.PathError{
```

Comment:
I'm wary of unnecessarily wrapping errors.
Do you have specific examples of `http.Filesystem` implementations that don't return an `os.PathError` when `Open()` fails?

The [issue](https://github.com/golang-migrate/migrate/issues/79) you've referenced existed before we added `PartialDriver`.

Reply:
> Do you have specific examples of http.Filesystem implementations that don't return an os.PathError when Open() fails?

I actually didn't check that before you asked but it turns out that yes, for example, `_escStaticFS` can return bare `os.ErrNotExist`.